server/server.go

```diff
@@ -63,11 +79,22 @@ func Start(config *rest.Config, discoveryInterval, informerRelist time.Duration,
 	// Create dynamic informer factory (for sharing dynamic informers).
 	dynInformers := dynamicinformer.NewSharedInformerFactory(dynClient, informerRelist)
 
+	// Create EventBroadcaster, for creating EventRecorders
+	broadcaster := record.NewBroadcaster()
+
+	clientSet, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to init clientSet: %w", err)
+	}
+	// TODO check server API version and record to either new or old events API
+	broadcaster.StartRecordingToSink(&typedcorev1.EventSinkImpl{Interface: clientSet.CoreV1().Events(metav1.NamespaceAll)})
```

Comment:
TODO: when run on older cluster, there might be earlier version of events API, need to add mechanism to recognize that. Also, need to check in which version of events were added